controller/lib/core/flow_integrator.h

```diff
@@ -0,0 +1,87 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "hal.h"
+#include "pid.h"
+#include "units.h"
+
+// Integrates flow over time to calculate volume.
+//
+// Also computes a constant correction to the flow that tries to account for
+// slow-changing error in the flow, based on certain points in time when we
+// know what the volume "should be".
+class FlowIntegrator {
+public:
+  FlowIntegrator();
+
+  // Updates the tidal volume given this flow.
+  //
+  // FlowIntegrator will correct this flow according to FlowCorrection() --
+  // don't pre-correct it yourself!
+  void AddFlow(Time now, VolumetricFlow uncorrected_flow);
+
+  Volume GetTV() const { return volume_; }
+
+  // This value is added to measured flow in an attempt to drive flow to the
```

Comment:
Please clarify what "is added" means - does that mean that the caller must compute flow as raw_flow + FlowCorrection()?